src/Microsoft.Azure.WebJobs/StorageAttribute.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Base class for all Azure Storage binding attributes.
+    /// </summary>
+    public abstract class StorageAttribute : Attribute
```

Comment:
StorageAccountAttribute is a class/method/parameter level attribute that as its name indicates is solely for specifying an account name. Rather than overload that, I think its cleaner to introduce an abstract base that is not specific to storage account and could have other shared properties in the future. It's just cleaner. Note that customers can't use this base attribute so it won't cause any issues/confusion - it is just for us to share common properties.

As an alternative, we could also introduce a "ConnectionSetting" attribute similar or deriving from AppSettingAttribute that can be applied to such properties. E.g.:

```csharp
public class QueueAttribute : Attribute
{
    [ConnectionSetting]
    public string Connection { get; set; }
}
```

That attribute would apply to both Storage and ServiceBus. The drawback there is that it would require reflection to access/set the value.